distribution/client/src/lib/storage.ts

```diff
@@ -82,39 +82,29 @@ export default class Storage {
     const pluginPath = this.pluginsDirectory();
     const retArray = [];
     plugins.forEach((plugin) => {
-      retArray.push(new Promise((resolve, reject) => {
-        fs.remove(`${pluginPath}/${plugin}`, (err) => {
-          if(err) {
-            logger.debug(`${pluginPath}/${plugin} was not found.`);
-          }
-          logger.info(`${pluginPath}/${plugin} was removed`);
-          resolve(true);
-        });
-      }));
-      retArray.push(new Promise((resolve, reject) => {
-
-        fs.unlink(`${pluginPath}/${plugin}.hpi`, (err) => {
-          if (err) {
-            logger.debug(`${pluginPath}/${plugin}.hpi was not found.`);
-          }
-          logger.info(`${pluginPath}/${plugin}.hpi was deleted`);
-          UI.publish(`Deleted ${plugin}.hpi`);
-          resolve(true);
-        });
-
-      }));
-      retArray.push(new Promise((resolve, reject) => {
-
-        fs.unlink(`${pluginPath}/${plugin}.jpi`, (err) => {
-          if (err) {
-            logger.debug(`${pluginPath}/${plugin}.jpi was not found.`);
-          }
-          logger.info(`${pluginPath}/${plugin}.jpi was deleted`);
-          UI.publish(`Deleted ${plugin}.jpi`);
-          resolve(true);
-        });
-
-      }));
+      retArray.push(
+        remove(`${pluginPath}/${plugin}`)
+          .then(() => {
+            logger.info(`${pluginPath}/${plugin} was deleted.`);
+            return true;
+          }).catch(() => logger.debug(`${pluginPath}/${plugin} was not found.`))
```

Comment:
This LGTM: we will to reintroduce the `error` parameter when we implement https://issues.jenkins-ci.org/browse/JENKINS-54207. AFAIU, this change does keep the current behavior as-is.

The current behavior is actually wrong, but the good news is that it be your next friend @jennyfive now you're on a roll ;-): if you want to give a try to https://issues.jenkins-ci.org/browse/JENKINS-54207, we'll be happy to help :)